kevm-pyk/src/kevm_pyk/kproj/evm-semantics/evm.md

```diff
@@ -608,6 +612,14 @@ After executing a transaction, it's necessary to have the effect of the substate
            ...
          </message>
 
+    rule <k> #finalizeTx(false => true) ... </k>
+         <useGas> false </useGas>
+         <txPending> ListItem(MsgId:Int) REST => REST </txPending>
+         <message>
+           <msgID> MsgId </msgID>
+           ...
+         </message>
+
```

Comment:
I guess, once again, are these rules important to duplicate for the performance benefits? I guess it makes thinsg more consistent, so maybe it's OK. It just means we have more duplicated code though.

Reply:
All duplicated code is clearly marked in terms of `<useGas>` --- if anyone is changing something that has `<useGas>` true, then they also have to change their `<useGas>`-false equivalent. I will check what I can do with the `#if`s.